vpc/main.tf

```diff
@@ -35,23 +35,23 @@ resource "aws_vpc" "main" {
 }
 
 resource "aws_eip" "nat" {
-  count = var.enable_eip ? local.max_subnet_length : 0
+  count = var.enable_eip ? local.nat_gateway_count : 0
   # checkov:skip=CKV2_AWS_19:EIP is used by NAT Gateway
   vpc = true
   tags = merge(local.common_tags, {
-    Name = "${var.name}-eip${count.index}"
+    Name = var.single_nat_gateway ? "${var.name}-eip" : "${var.name}-eip${count.index}"
```

Comment:
Any reason to not just use the same `${var.name}-eip${count.index}` naming?  I'm thinking just from a code simplicity standpoint.

Reply:
no reason, just didn't like the numbers when it would always be 1. But good point on the simplicity. I've reverted back to the count naming convention.